Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/BayesQuasi2.py

```diff
@@ -0,0 +1,306 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2023 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+# pylint: disable=invalid-name,too-many-instance-attributes,too-many-branches,no-init,redefined-builtin
+
+from mantid.api import AlgorithmFactory, MatrixWorkspaceProperty, PropertyMode, Progress
+from mantid.kernel import StringListValidator, Direction
+from mantid import logger
+from IndirectCommon import GetThetaQ
+
+from typing import Dict
+from numpy import ndarray
+import numpy as np
+
+from quickBayesHelper import QuickBayesTemplate
+
+try:
+    from quickBayes.utils.general import get_background_function
+except (Exception, Warning):
+    import subprocess
+
+    print(
+        subprocess.Popen(
+            "python -m pip install -U quickBayes==1.0.0b14",
+            shell=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            stdin=subprocess.PIPE,
+        ).communicate()
+    )
+    # used in BayesStretch, pip install here to prevent a seperate check
+    print(
+        subprocess.Popen(
+            "python -m pip install -U joblib",
```

Comment:
joblib is on conda-forge, so this can be added as a dependency in developer files and in the conda recipe

Reply:
could you point me at the files and I will update them

Reply:
I think you'd want to add it to the `run` section of the mantid recipe:
https://github.com/mantidproject/mantid/blob/main/conda/recipes/mantid/meta.yaml
and in each of the windows/linux/macos mantid-developer files.